lib/memory_diagnostics.rb

```diff
@@ -22,6 +22,7 @@ def self.compare(from=nil, to=nil)
     diff = diff.map do |id|
       ObjectSpace._id2ref(id) rescue nil
     end.compact!
```

Comment:
Actually wouldn't this always be `nil`? 

`compact!` never returns the object it seems.

```
[1, 2, 3, 1].compact!
=> nil
```